controllers/promiserelease_controller.go

```diff
@@ -196,43 +209,74 @@ func (r *PromiseReleaseReconciler) delete(o opts, promiseRelease *v1alpha1.Promi
 	return defaultRequeue, nil
 }
 
-func (r *PromiseReleaseReconciler) reconcileOnInstalledPromise(o opts, promiseRelease *v1alpha1.PromiseRelease) (ctrl.Result, error) {
+func (r *PromiseReleaseReconciler) promiseExistsAtDesiredVersion(o opts, promiseRelease *v1alpha1.PromiseRelease) (bool, error) {
 	promises := &v1alpha1.PromiseList{}
 	err := o.client.List(o.ctx, promises, client.MatchingLabels{
 		promiseReleaseNameLabel: promiseRelease.GetName(),
 	})
+
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to list promises: %w", err)
+		return false, fmt.Errorf("failed to list promises: %w", err)
 	}
 
 	switch len(promises.Items) {
 	case 1:
-		if !promises.Items[0].GetDeletionTimestamp().IsZero() {
-			return defaultRequeue, nil
-		}
-		if promises.Items[0].Labels[promiseVersionLabel] == promiseRelease.Spec.Version {
-			break
+		if promises.Items[0].Labels[v1alpha1.PromiseVersionLabel] == promiseRelease.Spec.Version {
+			return true, nil
 		}
 		fallthrough
 	case 0:
-		promiseRelease.Status.Installed = false
-		err = o.client.Status().Update(o.ctx, promiseRelease)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to update PromiseRelease status: %w", err)
+		return false, nil
+	default:
+		return false, fmt.Errorf("expected 0 or 1 promises, got %d", len(promises.Items))
+	}
+}
+
+func (r *PromiseReleaseReconciler) updateStatus(o opts, pr *v1alpha1.PromiseRelease,
```

Comment:
it looks like this method is updating the Installed condition + the object status.status; name could be a bit more descriptive